docs/guides/upstream-mtls.md

```diff
@@ -0,0 +1,158 @@
+---
+
+# Upstream mTLS With Pomerium
+
+Part of a complete zero-trust security model is secure communication between your authenticated proxy (Pomerium) and the upstream service it provides access to. This means both Pomerium *and* the upstream service will authenticate each other.
+
+Pomerium confirms the identity of the upstream service by the TLS certificate it serves. See [`tls_custom_ca_file`] and [`tls_server_name`] for more information on configuring Pomerium to accept an upstream's TLS certificate.
+
+How an upstream service is configured to validate and accept client certificates is entirely up to that specific service and how it's configured. To provide a general explanation, in this guide we will use [OpenSSL] and [mkcert] to demonstrate how to configure Pomerium to provide the client certificate.
+
+## Before You Begin
+
+- To simplify this proof-of-concept demonstration, we assume that Pomerium is running in all-in-one mode, configured to communicate with an identity provider (**IdP**), and all command are run on the same host. You may have to move files and/or adjust paths and commands to match a different configuration.
+
+    See [Install - Binaries](/docs/install/binary.md) for more information on initial setup.
+
+- This guide uses [mkcert] to provide certificates. mkcert is a development tool, and not designed for production environments. In this guide, mkcert takes the place of your trusted certificate tooling solution.
+
+- We use `*.pomerium.localhost.io` as the placeholder for the Pomerium domain space. If you're testing on your local computer you can edit `/etc/hosts` or use dnsmasq on [macOS](https://blog.thesparktree.com/local-development-with-wildcard-dns) or [Linux](https://sixfeetup.com/blog/local-development-with-wildcard-dns-on-linux) systems to route this domain space to your local computer.
+
+## Create Certificates
+
+!!!include(install-mkcert.md)!!!
+
+1. Create a certificate and key for our demo upstream service, OpenSSL:
+
+    ```bash{1}
+    mkcert openssl.localhost
+
+    Created a new certificate valid for the following names 📜
+    - "openssl.localhost"
+
+    The certificate is at "./openssl.localhost.pem" and the key at "./openssl.localhost-key.pem" ✅
+
+    It will expire on 10 March 2024 🗓
+    ```
+
+1. Create a client certificate and key for Pomerium to use:
+
+    ```bash{1}
+    mkcert -client 'pomerium@localhost'
+
+    Created a new certificate valid for the following names 📜
+    - "pomerium@localhost"
+
+    The certificate is at "./pomerium@localhost-client.pem" and the key at "./pomerium@localhost-client-key.pem" ✅
+
+    It will expire on 10 March 2024 🗓
+    ```
+
+1. Change ownership of the client certificate files and move them to Pomerium's configuration directory:
+
+    ```bash
+    sudo chown pomerium:pomerium pomerium@localhost-client*pem /etc/pomerium
+    sudo mv pomerium@localhost-client*pem /etc/pomerium/
+    ```
```

Comment:
I think we can skip the move step since file references can be relative.  Just use `./` instead of `/etc/pomerium`.  

```suggestion
```

Reply:
Per discussion, this is required for the assumed environment. Clarified by ed424ac6e4c0fe5aac926fafde4685e59fc915bb